src/gui/properties/speedplotview.h

```diff
@@ -0,0 +1,89 @@
+#ifndef SPEEDPLOTVIEW_H
+#define SPEEDPLOTVIEW_H
+
+#include <QGraphicsView>
+#include <QPen>
+#include <QQueue>
+#include <QMap>
+
```

Comment:
You may want to google for "c++ forward declaration".
First let's assume you move all the implementation code below into a .cpp file and leave this header clean.
Instead of doing here `#include <QPen>` you can do this simple thing: `class QPen;`  <-- this is the forward declaration
Then in the .cpp you must do `#include <QPen>`
NOTE: This isn't possible for all headers. Basic rules can be found [here](https://stackoverflow.com/a/553869)

From those 3 includes I think QPen, QQueue and QMap can be forward declared. If you get weird compilation errors. About this comment and I'll help you.